ADAL/tests/integration/ADAuthorityValidationIntegrationTests.m

```diff
@@ -104,16 +99,18 @@ - (void)testAadNormalFlow
     
     [ADTestURLSession addResponse:[ADTestURLResponse responseValidAuthority:authority]];
     
+    XCTestExpectation* expectation = [self expectationWithDescription:@"Validate valid authority."];
     [authorityValidation validateAuthority:requestParams
                            completionBlock:^(BOOL validated, ADAuthenticationError * error)
      {
          XCTAssertTrue(validated);
          XCTAssertNil(error);
          
-         TEST_SIGNAL;
+         [expectation fulfill];
      }];
     
-    TEST_WAIT;
+    [self waitForExpectationsWithTimeout:1 handler:nil];
```

Comment:
Regarding constant vs hardcoded value. I've read some "best practices" for UT long time ago, and the idea was that we **should** use hardcoded values in tests (can try to find it if it is interesting). There are several reasons. Firstly, if we change that constant latter, it will affect all tests and some of them might fail (so we have dependency here). Secondly, test should be treated as documentation for you code. If we have constant, developer needs to do extra step to find out what is the value of that constant. But, if you guys think that constant here is better, I don't mind to add it. Regarding @unpluggedk question. One second should be enough here because we don't do real networks calls, and moreover, because of **ADTestURLSession**, the code in block is invoked on main thread.